### DISCUSSION TITLE
How to use createFactory helper to define rails-style route controllers? (GET, POST, PUT, etc)?

### DISCUSSION THREAD

**Discussion by tconroy** [honojs/hono#3563] [Q&A]
Reading through [the docs](https://hono.dev/docs/helpers/factory#createfactory) it's recommended to use the `createFactory()` helper to create rails-style controllers / route handlers. 

However, the docs do not include examples of what this looks like.

The [best practices](https://hono.dev/docs/guides/best-practices#factory-createhandlers-in-hono-factory) also suggests using `createFactory()` for this purpose but does not include example of defining >1 handler with the factory.

Would someone please provide an example of how to create colocated handlers (rails-like) with full type safety? Also, when to use `factory.createHandlers` vs when to use `factory.createApp`?

**Comment by tconroy** [comment]
I have discovered this works, but unclear if best practice / maximum type safety.

```ts
// src/index.ts

import { Hono } from 'hono'
import { UserController } from './User.controller.ts'

const app = new Hono()

app.route('/user', UserController);

export default app
```

```ts
// src/User.controller.ts

import { createFactory } from 'hono/factory';
import { logger } from 'hono/logger';

export const UserController = createFactory().createApp()
  .get('/', logger(), (c) => { return c.text('Hello Hono!!!!!'); })
  .get('/:id', logger(), (c) => { return c.text(`hi lil user ${c.req.param('id')}`); });

```

**Reply by maou-shonen** [reply]
Your example code is not a controller,
but it is the hono.js recommend usage.
